### DISCUSSION TITLE
25ms minimum for change monitoring

### DISCUSSION THREAD

**Discussion by mocsa** [mpaperno/WASimCommander#22] [Q&A]
In the README.md, under "Subscribe to Calculator Code and Get Variable results", it says that "the monitoring rate can be configured at any rate down to the millisecond" but there is a 25ms minimum. Why is there a minimum? Where is this value coming from?

I guess, for monitoring change of variables WASimCommander uses the SimConnect_RequestDataOnSimObject function which has the period parameter which can be set to SIMCONNECT_PERIOD_SIM_FRAME to monitor the data every frame. This suggests that the effective monitoring rate depends on how many frames the user's PC can render every second. If the framerate is bigger than 40 fps then the monitoring rate will be less than 25ms. Why does WASimCommander have a 25ms limit?

**Comment by mpaperno** [comment]
Hi Csaba,

> Why is there a minimum?

This sets how often the variable value change checks runs on the Sim side. This limit is set/enforced on the WASimModule side and is meant to help ensure the sim itself isn't spending too much time in the module's code.

> Where is this value coming from?

It's set in [WASimCommander.h (`TICK_PERIOD_MS`)](https://github.com/mpaperno/WASimCommander/blob/next/src/include/WASimCommander.h#L65) and AFAICR is used in the module code only. Specifically in the [`tick()` function](https://github.com/mpaperno/WASimCommander/blob/next/src/WASimModule/WASimModule.cpp#L1522).

The choice of 25ms is somewhat arbitrary, 40Hz just seemed like "fast enough" for most cases and still prevents the module code from running constantly.

You're right that the maximum rate of any variable's value change is dictated by the sim's running speed.

Out of curiosity, which data point(s) are you interested in receiving at faster rates?

> I guess, for monitoring change of variables WASimCommander uses the SimConnect_RequestDataOnSimObject function ...

Not really, no...  The module itself reads the variable values (then compares for changes, if requested to, and sends out all updates), not SimConnect itself.  Because the original point of WASimCommander was to get to variables that can't be accessed via SimConnect in the first place. So on the module (sim) side, the check loop (`tick()` function) runs at a maximum of 40Hz which limits the refresh rate of any variable being read.

HTH,
-Max
